subprojects/core/src/main/java/org/gradle/caching/internal/BuildCacheServiceProvider.java

```diff
@@ -63,12 +63,12 @@ public BuildCacheService createBuildCacheService() {
 
         RoleAwareBuildCacheService buildCacheService;
         if (local.isEnabled()) {
-            if (remote != null && remote.isEnabled()) {
+            if (canUseRemoteBuildCache(remote) && !startParameter.isOffline()) {
                 buildCacheService = createDispatchingBuildCacheService(local, remote);
             } else {
                 buildCacheService = createStandaloneLocalBuildService(local);
             }
-        } else if (remote != null && remote.isEnabled()) {
+        } else if (canUseRemoteBuildCache(remote) && !startParameter.isOffline()) {
```

Comment:
Should we add some kind of message that we are disabling the remote build cache?